### PR TITLE
fix(pi): align with openpi & change image padding value

### PR DESCRIPTION
**Pull request by SkieyFly**
## Title

fix(pi): align with openpi & change image padding value

## Type / Scope

- **Type**: Bug

## Summary / Motivation

There are some misalignments between lerobot's and openpi's implementation in pi0/pi0.5 modeling. Besides, the image padding value is not fixed yet. So this PR is to
- Align implementation with openpi.
- Fix the bug of image padding value.

## Related issues

- Fixes / Closes: 
  - implementation alignment: #2863
  - image padding: #2218 #2512 #2650

## What changed

- remove loss truncation.
    - reference: openpi's [pi0_pytorch.py](https://github.com/Physical-Intelligence/openpi/blob/main/src/openpi/models_pytorch/pi0_pytorch.py#L373) & [train_pytorch.py](https://github.com/Physical-Intelligence/openpi/blob/main/scripts/train_pytorch.py#L529)
    - files: [modeling_pi05.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi05/modeling_pi05.py#L1255-L1257) & [modeling_pi0.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi0/modeling_pi0.py#L1282-L1284)
-  remove state padding before tokenization.
    - reference: openpi's [config.py](https://github.com/Physical-Intelligence/openpi/blob/main/src/openpi/training/config.py#L126-L138) & [config.py](https://github.com/Physical-Intelligence/openpi/blob/main/src/openpi/training/config.py#L139-L163)
    - file: [processor_pi05.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi05/processor_pi05.py#L71-L72)
- Change the image padding value to 0.0 for float32. I adopted the fixing approach as #2512, proposed by @Hukongtao.
    - files: [modeling_pi05.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi05/modeling_pi05.py#L192-L203) & [modeling_pi0.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi0/modeling_pi0.py#L194-L205) & [modeling_pi0_fast.py](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi0_fast/modeling_pi0_fast.py#L124-L135)

## How was this tested (or how to run locally)

manual test

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by jadechoghari** [comment]
@SkieyFly just for quick sanity check, if you run the local tests for pi0, do they all pass? 
https://github.com/huggingface/lerobot/tree/main/tests/policies/pi0_fast and https://github.com/huggingface/lerobot/tree/main/tests/policies/pi0_pi05

Thanks for the PR 🤗

**Comment by SkieyFly** [comment]
Hi @jadechoghari,
actually, the tests would fail (except for pi0), and this is expected, since the state is also padded before tokenization in the test, but I believe it is not the case in openpi's implementation (you can refer to the code I provided). 
However, when I later switch to the `main` branch to run the test (`test_pi05_original_vs_lerobot.py`), it still failed. Can you double check it for me? (maybe I misconfigured something 😢 
env: I'm using the latest `openpi` in tests

**Comment by jadechoghari** [comment]
@SkieyFly i see, do you have the bandwidth to evaluate the pi0/05 on libero dataset, evaluate it then make sure the results are similar to https://huggingface.co/docs/lerobot/en/libero#reproducing-%CF%80%E2%82%80%E2%82%85-results

File: src/lerobot/policies/pi05/modeling_pi05.py [hunk 1]
```diff
@@ -1252,10 +1252,6 @@ def forward(self, batch: dict[str, Tensor], reduction: str = "mean") -> tuple[Te
         # Compute loss (no separate state needed for PI05)
         losses = self.model.forward(images, img_masks, tokens, masks, actions)
 
-        # Truncate losses to actual action dimensions
-        original_action_dim = self.config.output_features[ACTION].shape[0]
-        losses = losses[:, :, :original_action_dim]
-
```
Comment (jadechoghari) [inline src/lerobot/policies/pi05/modeling_pi05.py:1258]: We still need to truncate the loss since the policy is outputting max_action_dim, however can you instead mask the loss?

Comment (SkieyFly) [reply src/lerobot/policies/pi05/modeling_pi05.py:1258]: Yes, the policy is outputting max_action_dim. However, pi doesn't seem to "truncate" the loss even though they pad to the max_action_dim as well. Maybe the final performance will not be affected greatly? 
I've been kind of occupied recently and I will find time to train and evaluate to see what the result is.

**Comment by xianglunkai** [comment]
@SkieyFly 
The current real-machine performance of lerobot-pi05 is significantly inferior to that of openpi-pi05, especially in terms of significant motion jitter. After carefully comparing the codes, I believe the main issue lies in the fact that openpi's prediction is relative to motion, while lerobot's is absolute. Could you please assist in making the necessary corrections and alignments. Thank you!

**Comment by Hukongtao** [comment]
Will this modification affect the performance of the base model?
https://huggingface.co/lerobot/pi05_base]?

**Comment by pkooij** [comment]
@xianglunkai thanks for this observation, although its not a core modeling difference it is a critical difference in how data is normalized and learned. I will add this as an option to LeRobot's pipeline
